modal/image.py

```diff
@@ -323,17 +361,18 @@ def _from_args(
         if build_function and len(base_images) != 1:
             raise InvalidError("Cannot run a build function with multiple base images!")
 
-        def _deps() -> List[_Object]:
-            deps: List[_Object] = list(base_images.values()) + list(secrets)
+        def _deps() -> Sequence[_Object]:
```

Comment:
type fixes - `list` is invariant and doesn't allow subtypes in the container typing-wise, so I changed it to a `Sequence` which is covariant (and used tuples internally for the same reason). a bit annoying